zerver/tests/test_upload.py

```diff
@@ -559,9 +560,17 @@ def check_xsend_links(name: Text, name_str_for_test: Text) -> None:
                 self.assertEqual(response['X-Accel-Redirect'],
                                  '/serve_uploads/../../'  + test_upload_dir +
                                  '/files/' + fp_path + '/' + name_str_for_test)
-
-        check_xsend_links('zulip.txt', 'zulip.txt')
-        check_xsend_links('áéБД.txt', '%C3%A1%C3%A9%D0%91%D0%94.txt')
+                if response.get('Content-disposition'):
```

Comment:
I think we probably want this conditional to be based on the test parameters, not what's in the response; that feels like a more robust test.  So e.g.

`if content_disposition != ''`

and then have an `else` statement to confirm that `attachment` is not in the `Content-disposition` in the request, I think?